src/Client/GuzzleFileFetcher.php

```diff
@@ -54,30 +56,37 @@ public static function createFromUri(string $baseUri) : self
     /**
      * {@inheritdoc}
      */
-    public function fetchFile(string $fileName, int $maxBytes):string
+    public function fetchFile(string $fileName, int $maxBytes): PromiseInterface
     {
-        try {
-            $response = $this->client->request('GET', $fileName);
-        } catch (ClientException $e) {
-            if ($e->getCode() === 404) {
-                throw new RepoFileNotFound("$fileName not found", 0, $e);
+        $onFulfilled = function (ResponseInterface $response) use ($fileName, $maxBytes) {
+            $body = $response->getBody();
+            $contents = $body->read($maxBytes);
+
+            // If we reached the end of the stream, we didn't exceed the maximum
+            // number of bytes.
+            if ($body->eof() === true) {
+                return $contents;
             } else {
```

Comment:
I personally like the "unncessary" `else` bits, because it makes the decision logic completely explicit, which to me seems like a safeguard against future bugs. That said, it's a stylistic thing so I'd be okay changing it.

Reply:
Removed the extra `else` blocks.